src/EnergyPlus/UnitVentilator.cc

```diff
@@ -3991,6 +3982,35 @@ namespace UnitVentilator {
         }
         return Residuum;
     }
+    
+    Real64 SetOAMassFlowRateForCoolingVariablePercent(Real64 const MinOAFrac,       // Minimum Outside Air Fraction
+                                                      Real64 const MassFlowRate,    // Design Outside Air Mass Flow Rate
+                                                      Real64 const MaxOAFrac,       // Maximum Outside Air Fraction
+                                                      Real64 const Tinlet,          // Inlet Temperature to Unit or Zone Temperature
+                                                      Real64 const Toutdoor         // Outdoor Air Temperature
+    )
+    {
+
+        Real64 ActualOAMassFlowRate(0.0);   // Result or return value
+        
+        if (Tinlet <= Toutdoor) {
+            
+            ActualOAMassFlowRate = MinOAFrac * MassFlowRate;
+            
+        } else { // Tinlet > Toutdoor
+            // Use cooler outside air to provide "free" cooling without over-cooling.
+            // First, use a simple load equals mass flow times Cp time Delta T equation to find OA Mass Flow Rate.
+            // Then, limit the OA Mass Flow Rate between the MinOA flow and the MaxOA flow.
+            
+            ActualOAMassFlowRate = std::abs(QZnReq) / (PsyCpAirFnWTdb(DataEnvironment::OutHumRat,Toutdoor) * (Tinlet-Toutdoor));
```

Comment:
Not sure why, but this is off by about 30W from what the cooling coil is doing. I'll attach a test idf that shows the cooling coil operating at about 30W whenever the OA flow is between min and max.
Oh - it's probably the fan heat?
Also, this probably needs to be sensitive to `OpMode == CycFanCycCoil` - then set to max?

Reply:
@mjwitte Before I make a change, what you are asking for is:

When OpMode == CycFanCycCoil, then set:

ActualOAMassFlowRate = MaxOAFrac * MassFlowRate

Correct?

Reply:
That was my initial thought. I'm fairly certain that's how the airloop economizer functions with cycling fan. Otherwise, you end up with two variables - fan runtime and OA fraction to mess with to meet the load. And I would expect you want to maximize the beneficial outdoor air.

Reply:
But I'm not sure setting OAMassFlowRate works - You want 100% OA that cycles on enough to meet the load.

Reply:
@mjwitte Elsewhere in this model, I am seeing the following code:

        if (FanOpMode == CycFanCycCoil) {
            OutAirMassFlowRate = min(OAMassFlowRate, Node(InletNode).MassFlowRate);
        }

which basically limits the outside air mass flow rate if the fan is cycling if I am understanding it correctly. Do you feel if I do something similar in the new function that this will resolve the concern?

Reply:
Not sure - this has to be done at the appropriate point in the process in order for the mixing (if any) and cycling control to all work from the correct value.

Reply:
But it seems like SimUnitVentOAMixer (which is where that reassignment that I showed in my previous comment) should be run and then it would take care of the cycling case?

Reply:
Hopefully, test files will tell . . .